.github/workflows/ruff-format.yml

```diff
@@ -0,0 +1,37 @@
+name: Checks and tests
+
+on:
+  push:  # Trigger on any branch push
+  pull_request:  # Trigger on pull request events
+    types: [opened, synchronize, reopened]
+  workflow_dispatch:  # Allow manual triggering
+
+jobs:
+  ruff-check:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Enable Debug Logging
+        run: echo "ACTIONS_STEP_DEBUG=true" >> $GITHUB_ENV
+
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+
+      - name: Install Ruff
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install ruff
+
+      #- name: Install Poetry and dependencies
+        #run: |
+          #pip install poetry
+          #poetry install --no-root --with develop
```

Comment:
I put that in but commented it out as it's only necessary if we need to actually run the code. To just do formatting there's no actual need for all the dependencies., and because it takes a minute or so to install everything it really slows stuff down.

If we want we can add a check to see if the dependency environment builds properly but as a separate task?

Reply:
I see. I have added this comment https://github.com/DataWaveProject/nonlocal_gwfluxes/pull/19#discussion_r1848725672 which explains another way to do it using `poetry`.

Reply:
fixed by using ruff action instead